Build of Ipopt fails on Debian Linux

Possibly a duplicate of #62, but I don't have the `lib64` folders.

It fails during the build of dependency Lapack:
```
INFO: Changing Directory to /home/rs/.julia/v0.4/Ipopt/deps/src/Ipopt-3.12.1/ThirdParty/Lapack/build
[...]
checking for pkg-config... no
checking whether -lblas has BLAS... no
checking for COIN-OR package Blas... skipped check via pkg-config, redirect to fallback
checking for COIN-OR package Blas (fallback)... no, dependency coinblas not available
configure: error: Required package BLAS not found.
```

But BLAS was built just before, and it seems successfully:
```
julia> Pkg.dir("Ipopt")
"/home/rs/.julia/v0.4/Ipopt"

shell> ls /home/rs/.julia/v0.4/Ipopt/
deps  doc  example  LICENSE.md  README.md  REQUIRE  src  test

shell> ls /home/rs/.julia/v0.4/Ipopt/deps
build.jl  downloads  src  usr

shell> ls /home/rs/.julia/v0.4/Ipopt/deps/usr
lib

shell> ls /home/rs/.julia/v0.4/Ipopt/deps/usr/lib/
libcoinblas.a  libcoinblas.la  pkgconfig

shell> ls /home/rs/.julia/v0.4/Ipopt/deps/usr/lib/pkgconfig/
coinblas.pc

shell> cat /home/rs/.julia/v0.4/Ipopt/deps/usr/lib/pkgconfig/coinblas.pc
prefix=/home/rs/.julia/v0.4/Ipopt/deps/usr
exec_prefix=${prefix}
libdir=${exec_prefix}/lib

Name: Blas
Description: Basic linear algebra package
URL: https://projects.coin-or.org/BuildTools
Version: 1.4.2
Libs: -L${libdir} -lcoinblas  -L/usr/lib/gcc/x86_64-linux-gnu/4.9 -L/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-linux-gnu/4.9/../../.. -lgfortran -lm -lquadmath
Cflags:
```

If this is a recent version of debian you could try using the builtin ipopt
package instead.

On Sep 9, 2016 09:31, "Robert Schwarz" <notifications@github.com> wrote:

> Possibly a duplicate of #62
> <https://github.com/JuliaOpt/Ipopt.jl/issues/62>, but I don't have the
> lib64 folders.
>
> It fails during the build of dependency Lapack:
>
> INFO: Changing Directory to /home/rs/.julia/v0.4/Ipopt/deps/src/Ipopt-3.12.1/ThirdParty/Lapack/build
> [...]
> checking for pkg-config... no
> checking whether -lblas has BLAS... no
> checking for COIN-OR package Blas... skipped check via pkg-config, redirect to fallback
> checking for COIN-OR package Blas (fallback)... no, dependency coinblas not available
> configure: error: Required package BLAS not found.
>
> But BLAS was built just before, and it seems successfully:
>
> julia> Pkg.dir("Ipopt")
> "/home/rs/.julia/v0.4/Ipopt"
>
> shell> ls /home/rs/.julia/v0.4/Ipopt/
> deps  doc  example  LICENSE.md  README.md  REQUIRE  src  test
>
> shell> ls /home/rs/.julia/v0.4/Ipopt/deps
> build.jl  downloads  src  usr
>
> shell> ls /home/rs/.julia/v0.4/Ipopt/deps/usr
> lib
>
> shell> ls /home/rs/.julia/v0.4/Ipopt/deps/usr/lib/
> libcoinblas.a  libcoinblas.la  pkgconfig
>
> shell> ls /home/rs/.julia/v0.4/Ipopt/deps/usr/lib/pkgconfig/
> coinblas.pc
>
> shell> cat /home/rs/.julia/v0.4/Ipopt/deps/usr/lib/pkgconfig/coinblas.pc
> prefix=/home/rs/.julia/v0.4/Ipopt/deps/usr
> exec_prefix=${prefix}
> libdir=${exec_prefix}/lib
>
> Name: Blas
> Description: Basic linear algebra package
> URL: https://projects.coin-or.org/BuildTools
> Version: 1.4.2
> Libs: -L${libdir} -lcoinblas  -L/usr/lib/gcc/x86_64-linux-gnu/4.9 -L/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-linux-gnu/4.9/../../.. -lgfortran -lm -lquadmath
> Cflags:
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/JuliaOpt/Ipopt.jl/issues/66>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABp0M4Mx3Sza0b6ZQhKAHRY9t98WUkS9ks5qoV_KgaJpZM4J5DWo>
> .
>

--- a/deps/build.jl
+++ b/deps/build.jl
@@ -44,7 +44,8 @@ provides(SimpleBuild,
             end
             @build_steps begin
                 ChangeDirectory(joinpath(srcdir,"ThirdParty","Lapack","build"))
-                `../configure --prefix=$prefix --disable-shared --with-pic`
+                `../configure --prefix=$prefix --disable-shared --with-pic
+                              --with-blas="$prefix/lib/libcoinblas.a -lgfortran"`
                 `make install`
             end
             @build_steps begin

